tests/test_socnav_planner_adapter.py

```
@@ -539,6 +539,86 @@ def _boom(self):
+    assert adapter._baseline_predictor is None
+
+
+def test_prediction_adapter_baseline_partial_miss_uses_constant_velocity_fallback() -> None:
+    """Per-pedestrian forecast misses should not become robot-position futures."""
+
+    class _Trajectory:
+        def __init__(self, mean: np.ndarray) -> None:
+            self.mean = mean
+
+    class _Prediction:
+        predictions = [_Trajectory(np.empty((0, 2), dtype=np.float32))]
```

Ruff (0.15.17)

[warning] Mutable default value for class attribute

(RUF012)